hack/verify-staging-meta-files.sh

```diff
@@ -18,9 +18,9 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 
-staging_repos=($(ls "${KUBE_ROOT}/staging/src/k8s.io/"))
+IFS=" " read -r -a staging_repos <<< "$(ls "${KUBE_ROOT}/staging/src/k8s.io/")"
```

Comment:
which shellcheck failure is this working around?

Reply:
This is the check output of `hack/verify-shellcheck.sh` :
```
In ./hack/verify-staging-meta-files.sh line 23:
staging_repos=($(ls "${KUBE_ROOT}/staging/src/k8s.io/"))
               ^-- SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
```

Reply:
`read` is more explicit and portable than `(<a delimited string>)`. we can't use `mapfile` due to macOS

Reply:
I'm not completely clear why we need to read this in as an array, see my comment below

Reply:
that's fair, but that is the previous behavior which imo was reasonable enough, as bash goes :^)
unless we're redesigning entirely, `($( ... ))` -> `read -r ...` is an improvement